### PR TITLE
[JAB] Created Failed User Journey U.I Test for Cat B+E

### PR DESCRIPTION
**Pull request by jono-basford-bjss**
## Description
Created the failed user journey for Cat b+e through the U.I. 
Created a helper file and a helper function to check for staleness of an element

### CODE REVIEW

File: test/e2e/features/cat-be/99-fulljourney.feature [hunk 1]
```diff
@@ -5,65 +5,122 @@ Feature: Full end to end journey for Cat B+E
    Scenario: Examiner completes a passed test with no faults
       Given I am logged in as "mobexaminer2" and I have a test for "Mr Callahan Eaton"
       When I check candidate details for "Mr Callahan Eaton"
-      And I start the test for "Mr Callahan Eaton"
-      And the candidate completes the declaration page
-      And the candidate confirms their declaration
+         And I start the test for "Mr Callahan Eaton"
```
Comment (andrew-lai-bjss) [inline test/e2e/features/cat-be/99-fulljourney.feature]: Can the indentation be in line with the other feature files

File: test/e2e/step-definitions/debrief-steps.ts [hunk 2]
```diff
@@ -36,6 +36,15 @@ When('I complete the pass details', () => {
   continuePassFinalisation();
 });
 
+When('I complete the fail details', () => {
+  const d255NoRadio = getElement(by.id('d255-no'));
+  clickElement(d255NoRadio);
+  const debreifWitnessedRadio = getElement(by.id('debrief-witnessed-yes'));
```
Comment (andrew-lai-bjss) [inline test/e2e/step-definitions/debrief-steps.ts]: Please correct constant name

File: test/e2e/features/cat-be/99-fulljourney.feature [hunk 3]
```diff
@@ -5,65 +5,122 @@ Feature: Full end to end journey for Cat B+E
...
-    And validation item "office-candidate-description-validation-text" should be "Describe the candidate"
-    And validation item "office-candidate-description-validation-text" should be visible
-    And validation item "office-weather-validation-text" should be "Select weather conditions"
-    And validation item "office-weather-validation-text" should be visible
-    When I complete the office write up
-    And I upload the test
-    Then I should see the "Journal" page
-    And the test result for "Mr Dixon Clayton" is "11"
+   Scenario: Examiner completes a failed test with various faults
+      Given I am logged in as "mobexaminer2" and I have a test for "Mr Dixon Clayton"
```
Comment (andrew-lai-bjss) [inline test/e2e/features/cat-be/99-fulljourney.feature]: It would be nicer and quicker for the UI tests if you used a different candidate in either this or the termination scenario.

File: e2e-simulator-bdd.conf.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ exports.config = {
   allScriptsTimeout: 11000,
   capabilities: {
     platformName: 'iOS',
-    platformVersion: '12.1',
```
Comment (andrew-lai-bjss) [inline e2e-simulator-bdd.conf.js]: Can you revert this change as it will break the build which runs on 12.1

**Review by andrew-lai-bjss — APPROVED** [review]

**Review by MarkPatt89 — APPROVED** [review]
